include/swift/AST/Expr.h

```diff
@@ -4021,11 +4021,14 @@ class DynamicTypeExpr : public Expr {
 /// subexpressions of that AST node.
 class OpaqueValueExpr : public Expr {
   SourceRange Range;
+  Expr *UnderlyingValue;
```

Comment:
`OpaqueValueExpr` is a general-purpose construct where the "underlying value" comes from something elsewhere in the AST, above the `OpaqueValueExpr`. Is there some other place we can put the original initializer value that doesn't require adding this field to `OpaqueValueExpr`? One option: create a new `Expr` node `ImplicitlyWrappedPropertyValueExpr`  that stores the `OpaqueValueExpr` and its underlying value. This will make `findWrappedValuePlaceholder` even simpler and avoid altering the design of `OpaqueValueExpr`.

Reply:
Yes, I think having a dedicated `Expr` node for the wrapped value is the right thing to do. I was also worried about something else starting to use the `underlyingValue` field that I added here and breaking my assumptions in this PR that it's only used for property wrappers. Thank you!